### PR TITLE
Added connection support in s3 sink

### PR DESCRIPTION
**Pull request by itsankit-google**
[PLUGIN-968](https://cdap.atlassian.net/browse/PLUGIN-968)

### CODE REVIEW

File: docs/S3-batchsink.md [hunk 1]
```diff
@@ -13,6 +13,11 @@ Properties
 ----------
 **Reference Name:** Name used to uniquely identify this sink for lineage, annotating metadata, etc.
 
+**Use Connection** Whether to use a connection, if a connection is used,
+the credentials does not need to be provided.
+
+**Connection** Name of the connection to use, should have the macro function ${conn:(connection-name)} to provide.
```
Comment (yaojiefeng) [inline docs/S3-batchsink.md]: Should be `${conn(connection-name)}`

File: src/main/java/io/cdap/plugin/aws/s3/sink/S3BatchSink.java [hunk 2]
```diff
@@ -164,22 +160,21 @@ public void validate() {
...
         }
       }
 
       if (!containsMacro(NAME_PATH) && !path.startsWith("s3a://") && !path.startsWith("s3n://")) {
         collector.addFailure("Path must start with s3a:// or s3n://.", null).withConfigProperty(NAME_PATH);
       }
 
-      if (!containsMacro(NAME_PATH) && path.startsWith("s3n://") && !Strings.isNullOrEmpty(sessionToken)) {
+      if (!containsMacro(NAME_PATH) && path.startsWith("s3n://") && connection != null &&
+      !Strings.isNullOrEmpty(connection.getSessionToken())) {
```
Comment (yaojiefeng) [inline src/main/java/io/cdap/plugin/aws/s3/sink/S3BatchSink.java]: nit - indentation

**Review by yaojiefeng — APPROVED** [review]
Just minor comments, LGTM

File: docs/S3-batchsink.md (hunk 1 above)
Comment (itsankit-google) [reply docs/S3-batchsink.md]: done

File: src/main/java/io/cdap/plugin/aws/s3/sink/S3BatchSink.java (hunk 2 above)
Comment (itsankit-google) [reply src/main/java/io/cdap/plugin/aws/s3/sink/S3BatchSink.java]: done

**Pull request merged**